internal/store/hpa.go

```diff
@@ -132,8 +132,8 @@ var (
 
 					for j, m := range metrics {
 						metric := m
-						metric.LabelKeys = []string{"condition", "status"}
-						metric.LabelValues = append([]string{string(c.Type)}, metric.LabelValues...)
+						metric.LabelKeys = []string{"condition", "status", "reason"}
+						metric.LabelValues = append(metric.LabelValues, string(c.Type), c.Reason)
```

Comment:
For controlling cardinality, It's better to have a known set of Reasons to iterate over. You can find a similar implementation in [pod.go](https://github.com/kubernetes/kube-state-metrics/blob/master/internal/store/pod.go#L37).

Reply:
I am going to remove the addition of `reason` and defer this to a separate PR since this is not related to utilization metrics anyway.